packages/ckeditor5-list/src/documentlist/documentlistediting.ts

```diff
@@ -767,72 +768,68 @@ function modelChangePostFixer(
-		} else {
-			selection = model.createSelection( selectable );
-		}
+		const selection = selectable ?
+			editor.model.createSelection( selectable ) :
+			editor.model.document.selection;
 
-		// Get a reference list item. Inserted list items will be fixed according to that item.
-		const pos = selection.getFirstPosition()!;
-		let refItem = null;
+		const position = selection.getFirstPosition()!;
 
-		if ( isListItemBlock( pos.parent ) ) {
-			refItem = pos.parent;
-		} else if ( isListItemBlock( pos.nodeBefore ) ) {
-			refItem = pos.nodeBefore;
-		}
+		// Get a reference list item. Indention and type of inserted list items will be fixed according to that item.
+		let refItem: ListElement;
 
-		// If there is `refItem` it means that we do insert list items into an existing list.
-		if ( !refItem ) {
-			return;
+		if ( isListItemBlock( position.parent ) ) {
+			refItem = position.parent;
+		} else if ( isListItemBlock( position.nodeBefore ) ) {
+			refItem = position.nodeBefore;
+		} else {
+			return; // Content is not copied into a list.
 		}
 
-		// First list item in `data` has indent equal to 0 (it is a first list item). It should have indent equal
-		// to the indent of reference item. We have to fix the first item and all of it's children and following siblings.
-		// Indent of all those items has to be adjusted to reference item.
-		const indentChange = refItem.getAttribute( 'listIndent' ) - item.getAttribute( 'listIndent' );
+		editor.model.change( writer => {
+			const refIndent = refItem.getAttribute( 'listIndent' );
+			const refType = refItem.getAttribute( 'listType' );
 
-		// Fix only if there is anything to fix.
-		if ( indentChange <= 0 ) {
-			return;
-		}
+			if ( isListItemBlock( content ) ) {
+				// Content is already a list.
+				const indentDiff = Math.max( refIndent - content.getAttribute( 'listIndent' ), 0 );
 
-		model.change( writer => {
-			// Adjust indent of all "first" list items in inserted data.
-			for ( const { node } of iterateSiblingListBlocks( item, 'forward' ) ) {
-				writer.setAttribute( 'listIndent', node.getAttribute( 'listIndent' ) + indentChange, node );
+				for ( const { node } of iterateSiblingListBlocks( content, 'forward' ) ) {
+					writer.setAttributes( {
+						listIndent: node.getAttribute( 'listIndent' ) + indentDiff,
+						listType: refType
+					}, node );
+				}
+			} else if ( content.is( 'documentFragment' ) ) {
+				// Content is pasted into the list and might not be a list itself.
+				const indentDiff = Math.max( refIndent - ( content.getChild( 0 )!.getAttribute( 'listIndent' ) as number || 0 ), 0 );
+
+				for ( const child of content.getChildren() ) {
+					const isListItem = isListItemBlock( child );
+
+					writer.setAttributes( {
+						listIndent: ( isListItem ? child.getAttribute( 'listIndent' ) : 0 ) + indentDiff,
+						listItemId: isListItem ? child.getAttribute( 'listItemId' ) : ListItemUid.next(),
+						listType: refType
+					}, child );
+				}
```

Comment:
This part looks strange. You should not rely on the `content` type. It's just a different way of passing parameters to this function. The logic that alters the content should be unified.